homeassistant/components/lamarzocco/calendar.py

```diff
@@ -0,0 +1,112 @@
+"""Calendar platform for La Marzocco espresso machines."""
+
+from collections.abc import Iterator
+from datetime import datetime, timedelta
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN
+from .entity import LaMarzoccoBaseEntity
+
+CALENDAR_KEY = "auto_on_off_schedule"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switch entities and services."""
+
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([LaMarzoccoCalendarEntity(coordinator, CALENDAR_KEY)])
+
+
+class LaMarzoccoCalendarEntity(LaMarzoccoBaseEntity, CalendarEntity):
+    """Class representing a La Marzocco calendar."""
+
+    _attr_translation_key = CALENDAR_KEY
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the next upcoming event."""
+        now = dt_util.now()
+
+        events = self._get_events(
+            start_date=now,
+            end_date=now + timedelta(days=7),  # only need to check a week ahead
+        )
+        if events:
+            return events[0]
+        return None
```

Comment:
Consider one-linter:
```suggestion
        return next(iter(events), None)
```

Reply:
I'll add it, although I'm not sure I like it more in termsof readability...